### ISSUE TITLE
'File name too long' when staging file via url

### ISSUE THREAD

**Issue by zichner** [nextflow-io/nextflow#2415]
## Bug report 

When staging a file that is available at an url with a long list of parameters, the error 'File name too long' is thrown.
This is for instance the case when using an https link that includes a long token. As soon as the part of the url after the last '/' is longer than 256 characters, the error occurs

### Expected behavior and actual behavior

The expected behavior is that the the staging of such files works (e.g., by allowing the user to specify the file name or shortening it automatically)

### Steps to reproduce the problem

Here is a reproducible example. The parameter token is not required for that specific file, however, it illustrates the problem. As soon as the file name including the parameter list exceeds the maximum file name length, the error occurs.

```
my_file = 'https://raw.githubusercontent.com/nextflow-io/nextflow/master/tests/data/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token'

process get_file {
    echo true

    input:
        path input_file from my_file

    script:
        """
        head ${input_file}
        """
}
```

### Program output 

```
Oct-27 16:40:48.185 [FileTransfer-thread-1] DEBUG nextflow.file.FilePorter - Copying foreign file https://raw.githubusercontent.com/nextflow-io/nextflow/master/tests/data/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token to work dir: /data/work/stage/ae/4360e33a8544c5993ab3534a4f9578/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token
Oct-27 16:40:48.409 [FileTransfer-thread-1] WARN  nextflow.file.FilePorter - Unable to stage foreign file: https://raw.githubusercontent.com/nextflow-io/nextflow/master/tests/data/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token (try 1) -- Cause: /data/work/stage/ae/4360e33a8544c5993ab3534a4f9578/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token: File name too long
java.nio.file.FileSystemException: /data/work/stage/ae/4360e33a8544c5993ab3534a4f9578/p1.fa?token=token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token_token: File name too long
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
        at java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:434)
        at java.nio.file.Files.newOutputStream(Files.java:216)
        at java.nio.file.Files.copy(Files.java:3016)
        at nextflow.file.CopyMoveHelper.copyFile(CopyMoveHelper.java:124)
        at nextflow.file.CopyMoveHelper.copyToForeignTarget(CopyMoveHelper.java:205)
        at nextflow.file.FileHelper.copyPath(FileHelper.groovy:895)
        at nextflow.file.FilePorter$FileTransfer.stageForeignFile0(FilePorter.groovy:288)
        at nextflow.file.FilePorter$FileTransfer.stageForeignFile(FilePorter.groovy:257)
        at nextflow.file.FilePorter$FileTransfer.run(FilePorter.groovy:243)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### Environment 

* Nextflow version: 21.04.2 
* Java version: Groovy 3.0.7 on OpenJDK 64-Bit Server VM 1.8.0_272-b10
* Operating system: Linux
* Bash version: GNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)

**Comment by pditommaso** [comment]
Got your point. Actually stripping everything after the `?` could be reasonable.

**Comment by zichner** [comment]
Thanks for the fix of this issue!
It seems that it is not included in any of the "edge" releases so far. Will it still be included in the next stable release (v22.04.0)?
Thanks!

**Comment by pditommaso** [comment]
This is going to be released both with 22.03.0-edge and 22.04.0 stable

**Comment by zichner** [comment]
Very good. Thank you very much!
